server/graphql/v2/object/Host.js

```diff
@@ -590,14 +668,68 @@ export const Host = new GraphQLObjectType({
           if (dateRange) {
             where.createdAt = dateRange;
           }
+          let collectiveIds;
           if (args.account) {
             const collectives = await fetchAccountsWithReferences(args.account, { throwIfMissing: true });
             const collectiveIds = collectives.map(collective => collective.id);
             where.CollectiveId = { [Op.in]: collectiveIds };
           }
+
+          const expenseAmountOverTime = async () => {
+            let expenseAmountOverTime;
+            if (args.timeUnit) {
+              const dateFrom = args.dateFrom ? moment(args.dateFrom).toISOString() : undefined;
+              const dateTo = args.dateTo ? moment(args.dateTo).toISOString() : undefined;
+              expenseAmountOverTime = await sequelize.query(
+                `SELECT DATE_TRUNC(:timeUnit, "createdAt") AS "date", sum(amount) AS "amount", "currency"
+                 FROM "Transactions"
+                 WHERE kind = 'EXPENSE'
+                   AND "HostCollectiveId" = :hostCollectiveId
+                   AND type = 'DEBIT'
+                   ${collectiveIds ? `AND "CollectiveId" IN (:collectiveIds)` : ``}
+                   ${dateFrom ? `AND "createdAt" >= :dateFrom` : ``}
+                   ${dateTo ? `AND "createdAt" <= :dateTo` : ``}
+                 GROUP BY DATE_TRUNC(:timeUnit, "createdAt"), "currency"
+                 ORDER BY DATE_TRUNC(:timeUnit, "createdAt")`,
+                {
+                  type: sequelize.QueryTypes.SELECT,
+                  replacements: {
+                    hostCollectiveId: host.id,
+                    timeUnit: args.timeUnit,
+                    collectiveIds,
+                    dateFrom,
+                    dateTo,
+                  },
+                },
```

Comment:
The query seems 99% similar to the other one, only the `kind` changes. Could we extract that somewhere? Maybe a `getTransactionsTimeseries` in `server/lib/queries.js`?

Reply:
I've created a new function, `getTransactionsTimeseries` in `server/lib/queries.js` and move this query there. 👍🏽